### PR TITLE
Implements Previous Orders

### PR DESCRIPTION
**Pull request by rgbatty**
Visiting the Orders page as a user now displays links to each individual order

Order pages display:
- the time they were placed
- the status
- the timestamp for the current status
- the items purchased, their quantity, and subtotal
- a total cost for the cart

### CODE REVIEW

File: .byebug_history [hunk 1]
```diff
@@ -1,4 +1,22 @@
 exit
+Order.all
```
Comment (ErinGreenhalgh) [inline .byebug_history:2]: hmm, why is byebug history suddenly showing up? shouldn't this be in .gitignore?

File: app/models/order.rb [hunk 2]
```diff
@@ -4,13 +4,36 @@ class Order < ActiveRecord::Base
...
+    self.finished_at = DateTime.now
+    self.status = "cancelled"
+    save
+  end
+
+  def complete
+    self.finished_at = DateTime.now
+    self.status = "completed"
+    save
+  end
```
Comment (ErinGreenhalgh) [inline app/models/order.rb:26]: looks good

File: app/views/orders/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<h1>Order Details</h1>
+<h4> Placed At: <%= @order.created_at %> </h2>
+
+<table class="table">
+  <th>Order Status</th>
+  <th>Order Placed At</th>
+
+  <tr>
+    <td><%= @order.status %></td>
+    <td><%= @order.get_relevant_time %></td>
```
Comment (ErinGreenhalgh) [inline app/views/orders/show.html.erb:10]: i think we'll want to show the created at time in addition to the the cancelled or completed time

File: app/views/shared/_item_list.html.erb [hunk 4]
```diff
@@ -0,0 +1,31 @@
+<div class="row">
+  <div class="col-md-12" id = "title">
+    <h2>All Items</h2>
+  </div>
+</div>
+
+<% @items.each_with_index do |item, index| %>
+  <% if index % 3 == 0 && index != 0 %>
```
Comment (ErinGreenhalgh) [inline app/views/shared/_item_list.html.erb]: hmm, maybe we can state this is a more straightforward way; i assume this is to make the right number of items show up in the columns in our grid?

File: app/views/orders/show.html.erb (hunk 3 above)
Comment (rgbatty) [inline app/views/orders/show.html.erb:10]: Already there on line 2 :)

File: .byebug_history (hunk 1 above)
Comment (rgbatty) [inline .byebug_history:2]: Should be resolved with the fixed git_ignore. Was an artifact of a bad merge.

**Pull request merged**